runtime/debugger.c

```diff
@@ -96,7 +97,7 @@ static struct channel * dbg_out;/* Output channel on the socket */
 
 static char *dbg_addr = NULL;
 
-static struct ext_table breakpoints_table;
+static struct skiplist event_points_table = SKIPLIST_STATIC_INITIALIZER;
```

Comment:
I'm not sure where we are with restarting the runtime.  Since #71 the runtime can be unloaded cleanly, that is, all the memory it allocated dynamically is freed.  This will work for skiplists too, because the cells are allocated with `caml_stat_alloc`.  But I don't think you can just restart the runtime system, because a bunch of static initializations need to be redone.

Reply:
I think @murmour's intention was always to able to restart the runtime, although doing this with the debugger was probably not!

Reply:
No, @xavierleroy is right. See `startup_aux.c`:

```C
  if (shutdown_happened == 1)
    caml_fatal_error("caml_startup was called after the runtime "
                     "was shut down with caml_shutdown");
```

Reply:
Right.  @murmour called it "the unloadable runtime", not "the restartable runtime" :-)  

I think the scenario he had in mind is that the OCaml runtime and user code is in a DLL, and we want to release all memory used by OCaml before unloading the DLL.  The DLL could be reloaded in the same process, but then static initialization of globals would be performed again.

For a restartable runtime, we would need to rely less on static initialization of global variables and have more "init" functions.